lib/matplotlib/colorbar.py

```diff
@@ -449,14 +444,14 @@ def draw_all(self):
         if self.filled:
             self._add_solids(X, Y, C)
 
+    def set_norm(self, norm):
```

Comment:
I don't think that we want to have the colorbar in any way controlling the mappable.

Reply:
OK, I somewhat agree.  OTOH, right now, the colorbar is inexplicably subclassed from `ScalarMappable` which is where `set_norm` comes from.  I have no idea why that is or how to extricate it, but I don't think we should have a `colorbar.set_norm` that does nothing.

Reply:
... so actually making it *not* be a subclass of `ScalarMappable` *almost* works.  Is there a real reason this is a problem?

Reply:
I think the subclassing was just an unfortunate design choice on my part, long ago.  I suspect my thought was that it should be possible to make a colorbar without any accompanying mappable, so that was the starting point.  But I'm sure that can still be done in a straightforward way without subclassing.